tests/unit/Diff/Internal/AliasGroupListPatcherTest.php

```diff
@@ -0,0 +1,229 @@
+				) )
+			),
+			'remove some aliases is no-op (atomic)' => array(
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar', 'baz' ) )
+				) ),
+				new Diff( array(
+					'en' => new DiffOpRemove( array( 'foo', 'bar' ) )
+				) ),
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar', 'baz' ) )
+				) )
+			),
+			'add an existing language is no-op' => array(
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar' ) )
+				) ),
+				new Diff( array(
+					'en' => new Diff( array( new DiffOpAdd( 'baz' ) ) )
+				) ),
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar', 'baz' ) )
+				) )
+			),
+			'add an existing language is no-op (atomic)' => array(
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar' ) )
+				) ),
+				new Diff( array(
+					'en' => new DiffOpAdd( array( 'baz' ) )
+				) ),
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar' ) )
+				) )
+			),
+			'add two alias groups' => array(
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar' ) )
+				) ),
+				new Diff( array(
+					'de' => new Diff( array( new DiffOpAdd( 'foo' ), new DiffOpAdd( 'baz' ) ) ),
+					'nl' => new Diff( array( new DiffOpAdd( 'bar' ), new DiffOpAdd( 'baz' ) ) )
+				) ),
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar' ) ),
+					new AliasGroup( 'de', array( 'foo', 'baz' ) ),
+					new AliasGroup( 'nl', array( 'bar', 'baz' ) )
+				) )
+			),
+			'add two alias groups (atomic)' => array(
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar' ) )
+				) ),
+				new Diff( array(
+					'de' => new DiffOpAdd( array( 'foo', 'baz' ) ),
+					'nl' => new DiffOpAdd( array( 'bar', 'baz' ) )
+				) ),
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar' ) ),
+					new AliasGroup( 'de', array( 'foo', 'baz' ) ),
+					new AliasGroup( 'nl', array( 'bar', 'baz' ) )
+				) )
+			),
+			'remove a not existing language is no-op' => array(
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar' ) )
+				) ),
+				new Diff( array(
+					'de' => new Diff( array( new DiffOpRemove( 'bar' ) ) )
+				) ),
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar' ) )
+				) )
+			),
+			'remove a not existing language is no-op (atomic)' => array(
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar' ) )
+				) ),
+				new Diff( array(
+					'de' => new DiffOpRemove( array( 'bar' ) )
+				) ),
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo', 'bar' ) )
+				) )
+			),
+			'change different aliases is no-op' => array(
+				new AliasGroupList( array(
+					new AliasGroup( 'en', array( 'foo' ) )
+				) ),
+				new Diff( array(
+					'en' => new Diff( array( new DiffOpChange( 'bar', 'baz' ) ) )
```

Comment:
All test cases that are nested like this (`Diff` → `Diff` → `DiffOpChange`) should be duplicated, one with `Diff` → `Diff( …, true )` → `DiffOpChange` and one with `Diff` → `Diff( …, false )` → `DiffOpChange`.